Detectors/TRD/macros/convertRun2ToRun3Digits.C

```diff
@@ -0,0 +1,247 @@
+      reader = (AliRawReader*)readerDate;
+
+    } else if (rawDataInPath.Contains(":")) {
+      cout << "[I] Reading DATE monitoring events" << endl;
+      AliRawReaderDateOnline* readerRoot = new AliRawReaderDateOnline(rawDataInPath);
+      readerRoot->SelectEquipment(0, 1024, 1041);
+      readerRoot->Select("TRD");
+      //readerRoot->SelectEvents(7);
+      reader = (AliRawReader*)readerRoot;
+    }
+
+    AliTRDdigitsManager* digitMan = new AliTRDdigitsManager;
+    digitMan->CreateArrays();
+
+    AliTRDrawStream* rawStream = new AliTRDrawStream(reader);
+
+    TClonesArray trkl("AliTRDtrackletMCM");
+    rawStream->SetTrackletArray(&trkl);
+
+    int ievent = 0;
+    while (reader->NextEvent()) {
+      ievent++;
+      int eventtime = ievent * 12;
+
+      if (ievent >= nRawEvents)
+        break;
+
+      //digitMan->ResetArrays();
+
+      if (ievent % 10 == 0) {
+        cout << "Event " << ievent << endl;
+      }
+
+      // hntrkl->Fill(trkl.GetEntries());
+      while (rawStream->NextChamber(digitMan) >= 0) {
+        //hptphase->Fill(digMan->GetDigitsParam()->GetPretriggerPhase());
+      }
+
+      for (int det = 0; det < 540; det++) {
+        AliTRDSignalIndex* idx = digitMan->GetIndexes(det);
+
+        if (!idx)
+          continue;
+        if (!idx->HasEntry())
+          continue;
+
+        AliTRDarrayADC* digits;
+        digits = digitMan->GetDigits(det);
+        digits->Expand();
+
+        int row, col;
+        while (idx->NextRCIndex(row, col)) {
+          int tbsum = 0;
+          ArrayADC adctimes;
+          for (int timebin = 0; timebin < digitMan->GetDigits(det)->GetNtime(); timebin++) {
+            int adc = digitMan->GetDigits(det)->GetData(row, col, timebin);
+            hAdc->Fill(adc);
+            tbsum += adc;
+
+            adctimes[timebin] = digits->GetData(row, col, timebin);
+          }
+
+          run3Digits.push_back(o2::trd::Digit(det, row, col, adctimes, eventtime));
+
+          hTBsum->Fill(tbsum);
+        }
+      }
+      trkl.Clear();
+    }
+
+    delete rawStream;
+    if (reader)
+      delete reader;
+  }
+
+  // convert run2 digits if path set
+  if (run2DigitsInPath != "") {
+    cout << "Converting Run2 digits..." << endl;
+
+    TFile run2DigitsFile(run2DigitsInPath);
+    AliTRDdigitsManager* digitMan = new AliTRDdigitsManager;
+    digitMan->CreateArrays();
+
+    TIter next(run2DigitsFile.GetListOfKeys());
+    while (TObject* obj = next()) {
+      cout << "Processing " << obj->GetName() << endl;
+
+      // eventTime needs to be some increasing integer
+      string eventNumber(obj->GetName(), 5, 3);
+      int eventTime = stoi(eventNumber) * 12;
+
+      TTree* tr = (TTree*)run2DigitsFile.Get(Form("%s/TreeD", obj->GetName()));
+
+      for (int det = 0; det < 540; det++) {
+        digitMan->ClearArrays(det);
+        digitMan->ClearIndexes(det);
+      }
+
+      digitMan->ReadDigits(tr);
+
+      for (int det = 0; det < 540; det++) {
```

Comment:
You could replace 540 with AliTRDCommonParam::kNdet and a few other places where looping over 540